v3/unit_context.go

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package work
+
+import "go.uber.org/zap"
+
+// UnitContext represents context about a work unit passed through to work unit actions.
+type UnitContext struct {
```

Comment:
can't decide quite yet if it should be `UnitContext`, or `UnitActionContext`.

Reply:
after giving this more thought, even though the context initially appears to be about the work unit, the true encapsulation here is context required for the action to execute. much of the action context may include information about the work unit, but my belief is that the action context over time will diverge to include whatever data necessary to execute the action effectively, regardless if that data is about the unit or not.

for example, say we went down the route of executing actions even in the event the milestone they are attached to didn't succeed. we would likely want to pass along some error information into the action. this error isn't really unit information - it's parallel data that was generated while processing the unit, but it isn't data about the unit itself.